src/core/control/jobs/RenderJob.cpp

```diff
@@ -50,13 +50,7 @@ void RenderJob::rerenderRectangle(Rectangle<double> const& rect) {
 
     renderToBuffer(newMask.get());
 
-    std::lock_guard lock(this->view->drawingMutex);
-    if (!view->buffer.isInitialized()) {
-        // Todo: the buffer must not be uninitializable here, either by moving it into the job or by locking it at job
-        // creation a shared prt may also be suffice.
-        XOJ_CPP20_UNLIKELY return;
-    }
-    newMask.paintTo(view->buffer.get());
+    newMask.paintTo(buffer.lock()->mask.get());
```

Comment:
This can't happen, since buffer is a `SharedMask`, a copy of `view->buffer`. `SharedMask` owns shared smart pointers to `cairo_mask` the `std::mutex` and the data of the Mask. So they won't be deleted.